app/integrations.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"sort"
+	"strings"
+
+	"github.com/mattermost/mattermost-server/v6/app/request"
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+)
+
+type ListedApp struct {
+	Manifest struct {
+		AppID       string `json:"app_id"`
+		DisplayName string `json:"display_name"`
+		Version     string `json:"version"`
+	}
+
+	Installed bool `json:"installed"`
+	Enabled   bool `json:"enabled"`
+}
+
+func (a *App) GetInstalledIntegrations() ([]*model.InstalledIntegration, *model.AppError) {
+	return a.ch.getInstalledIntegrations()
+}
+
+func (ch *Channels) getInstalledIntegrations() ([]*model.InstalledIntegration, *model.AppError) {
+	out := []*model.InstalledIntegration{}
+
+	pluginsEnvironment := ch.GetPluginsEnvironment()
+	if pluginsEnvironment == nil {
+		return out, nil
+	}
+
+	plugins, err := pluginsEnvironment.Available()
+	if err != nil {
+		return nil, model.NewAppError("getInstalledIntegrations", "app.plugin.sync.read_local_folder.app_error", nil, err.Error(), 0)
+	}
+
+	for _, p := range plugins {
+		ignore := false
+		for _, id := range model.InstalledIntegrationsIgnoredPlugins {
+			if p.Manifest.Id == id {
+				ignore = true
+				break
+			}
+		}
+
+		if !ignore {
+			integration := &model.InstalledIntegration{
+				Type:    "plugin",
+				ID:      p.Manifest.Id,
+				Name:    p.Manifest.Name,
+				Version: p.Manifest.Version,
+				Enabled: pluginsEnvironment.IsActive(p.Manifest.Id),
+			}
+
+			out = append(out, integration)
+		}
+	}
+
+	if pluginsEnvironment.IsActive("com.mattermost.apps") {
+		enabledApps, appErr := ch.getInstalledApps()
+		if appErr != nil {
+			ch.srv.Log.Warn("Failed to fetch installed Apps", mlog.Err(appErr))
+			enabledApps = []ListedApp{}
+		}
+
+		for _, ap := range enabledApps {
+			ignore := false
+			for _, integration := range out {
+				if integration.ID == ap.Manifest.AppID {
+					integration.Type = "plugin-app"
+					ignore = true
+				}
+			}
+
+			if !ignore {
+				integration := &model.InstalledIntegration{
+					Type:    "app",
+					ID:      ap.Manifest.AppID,
+					Name:    ap.Manifest.DisplayName,
+					Version: ap.Manifest.Version,
+					Enabled: ap.Enabled,
+				}
+
+				out = append(out, integration)
+			}
+		}
+	}
+
+	// Sort result alphabetically, by display name.
+	sort.SliceStable(out, func(i, j int) bool {
+		return strings.ToLower(out[i].Name) < strings.ToLower(out[j].Name)
+	})
+
+	return out, nil
+}
+
+func (ch *Channels) getInstalledApps() ([]ListedApp, *model.AppError) {
+	rawURL := "/plugins/com.mattermost.apps/api/v1/marketplace"
+	values := url.Values{
+		"include_plugins": []string{"true"},
```

Comment:
Why are plugin apps included if the list of plugins is also fetched in `getInstalledIntegrations`?